plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -406,157 +321,6 @@ public void testNaNPartition()
-                session,
-                "create table partition_test10(\n"
-                        + "id integer,\n"
-                        + "a varchar,\n"
-                        + "b varchar,\n"
-                        + "ds varchar)"
-                        + "WITH (format='PARQUET')");
-        assertUpdate(session, "insert into partition_test9(id,a,ds) values(1, 'a','a')", 1);
-        assertUpdate(session, "insert into partition_test10(id,a,ds) values(1, 'a','a')", 1);
-        assertQuery(session, "select a.id, b.id from partition_test9 a join partition_test10 b on (a.id = b.id) where a.ds = 'a'", "SELECT 1, 1");
-        assertUpdate(session, "DROP TABLE partition_test9");
-        assertUpdate(session, "DROP TABLE partition_test10");
-    }
-
-    @Test
-    public void testPartitionPredicateAllowed()
-    {
-        Session session = Session.builder(getQueryRunner().getDefaultSession())
-                .setIdentity(Identity.forUser("hive")
-                        .withRole("hive", new SelectedRole(ROLE, Optional.of("admin")))
-                        .build())
-                .setCatalogSessionProperty("hive", "query_partition_filter_required", "true")
-                .build();
-
-        assertUpdate(
-                session,
-                "create table partition_test11(\n"
-                        + "id integer,\n"
-                        + "a varchar,\n"
-                        + "b varchar,\n"
-                        + "ds varchar)"
-                        + "WITH (format='PARQUET', partitioned_by = ARRAY['ds'])");
-        assertUpdate(session, "insert into partition_test11(id,a,ds) values(1, '1','1')", 1);
-        String query = "select id from partition_test11 where cast(ds as integer) = 1";
-        assertQuery(session, query, "select 1");
-        computeActual(session, "explain " + query);
-        assertUpdate(session, "DROP TABLE partition_test11");
-    }
-
-    @Test
-    public void testNestedQueryWithInnerPartitionPredicate()
-    {
-        Session session = Session.builder(getQueryRunner().getDefaultSession())
-                .setIdentity(Identity.forUser("hive")
-                        .withRole("hive", new SelectedRole(ROLE, Optional.of("admin")))
-                        .build())
-                .setCatalogSessionProperty("hive", "query_partition_filter_required", "true")
-                .build();
-
-        assertUpdate(
-                session,
-                "create table partition_test12(\n"
-                        + "id integer,\n"
-                        + "a varchar,\n"
-                        + "b varchar,\n"
-                        + "ds varchar)"
-                        + "WITH (format='PARQUET', partitioned_by = ARRAY['ds'])");
-        assertUpdate(session, "insert into partition_test12(id,a,ds) values(1, '1','1')", 1);
-        String query = "select id from (select * from partition_test12 where cast(ds as integer) = 1) where cast(a as integer) = 1";
```

Comment:
is there a reason to remove this testcase?

Reply:
wow, you did really good review here!

`// partition predicate in outer query only` test case covers inner/outer query where part filter is on the outer query
before my change, there wasn't such case and i think it was just a typo -- testing inner/outer query where part filter is on the inner query is not that interesting.

Reply:
> part filter is on the inner query is not that interesting

I see, so the "outer" case more interesting than "inner" one, because it'd be useful to see whether the planchecker is prematurely applied (on inner query first/separately without seeing part filter)